jsonify_stored/models/jsonify_stored_mixin.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2022 Camptocamp SA (http://www.camptocamp.com).
+# @author Simone Orsi <simone.orsi@camptocamp.com>
+# @author Matthieu Méquignon <matthieu.mequignon@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import json
+import logging
+from collections import namedtuple
+from itertools import groupby
+
+from psycopg2.extensions import AsIs
+
+from odoo import api, fields, models
+from odoo.tools.misc import get_lang, split_every
+
+from odoo.addons.base_sparse_field.models.fields import Serialized
+
+GrouperAndSorter = namedtuple("GrouperAndSorter", "exporter_id lang_code")
+
+_logger = logging.getLogger(__file__)
+
+
+class JsonifyStoredMixin(models.AbstractModel):
+    """This mixin aims to provide the ability to store jsonified data on model."""
+
+    _name = "jsonify.stored.mixin"
+    _description = "JSONify stored mixin"
+
+    # TODO: writeable, stored, compute default value, filter by current model?
+    ir_export_id = fields.Many2one("ir.exports", compute="_compute_ir_export_id")
+    jsonified_display = fields.Text(
+        compute="_compute_jsonified_display",
+    )
+    jsonified_data = Serialized(compute="_compute_jsonified_data", store=True)
+
+    def _compute_ir_export_id(self):
+        for record in self:
+            record.ir_export_id = record._jsonify_get_exporter()
+
+    def _jsonify_get_exporter(self):
+        """Hook to get the exporter to use."""
+        return False
+
+    def _compute_jsonified_display(self):
+        for record in self:
+            record.jsonified_display = json.dumps(
+                record.jsonified_data, sort_keys=True, indent=4
+            )
+
+    def _compute_jsonified_data(self):
+        """Compute JSON data for current recordset.
+
+        Optimized to always group by lang and parser
+        even when large mixed recordsets are updated.
+        """
+        default_lang = self._get_default_lang()
+        for grouper, group in self._compute_jsonified_data_groupby():
+            records = self.browse([r.id for r in group])
+            exporter = grouper.exporter_id
+            if not exporter:
+                records.jsonified_data = False
+                _logger.error("%s have no exporter", str(records._name))
+                continue
+            lang_code = grouper.lang_code or default_lang
+            # `get_json_parser` is cached
+            parser = exporter.get_json_parser()
+            self._update_jsonified_data(records.with_context(lang=lang_code), parser)
+
+    def _compute_jsonified_data_groupby(self):
+        _grouper_and_sorter = self._jsonify_data_grouper_and_sorter
+        return groupby(self.sorted(_grouper_and_sorter), key=_grouper_and_sorter)
+
+    def _get_default_lang(self):
+        return get_lang(self.env)
+
+    @staticmethod
+    def _jsonify_data_grouper_and_sorter(rec):
+        lang_code = None
+        if "lang_id" in rec._fields:
```

Comment:
Maybe get the `"lang_id"` field name from a property.
I understand it's a standard but maybe some model uses a different field for the lang field

Reply:
I don't think we need this. We are defining a convention here which most of times will be satisfied as this is the most common name for relations to languages. If you want to change it, you override this. To ease it I could put the lang in its own class prop or method

Reply:
Noted 👍🏻 Just to keep in mind though, there's also the "standard" `lang` Selection field (for example in `res.partner`)

Reply:
I let the implementer handle this on his/her model if wants to work w/ it :stuck_out_tongue_winking_eye: